Mathlib/NumberTheory/FactorisationProperties.lean

```diff
@@ -0,0 +1,193 @@
+  relations with prime and perfect numbers.
+
+## Main Definitions
+* `Nat.Abundant`: a natural number `n` is _abundant_ if the sum of its proper divisors are greater
+  than `n`
+* `Nat.Pseudoperfect`: a natural number `n` is _pseudoperfect_ if a subset of its proper divisors
+  equals `n`
+* `Nat.Deficient`: a natural number `n` is _deficient_ if the sum of its proper divisors is less
+  than `n`
+* `Nat.Weird`: a natural number is _weird_ if it is both abundant and not pseudoperfect
+
+## Main Results
+* `Nat.deficient_or_perfect_or_abundant`: A positive natural number is either deficient,
+  perfect, or abundant.
+* `Nat.prime_deficient`: All prime natural numbers are deficient.
+* `Nat.exists_infinite_deficients`: There are infinitely many deficient numbers.
+* `Nat.prime_pow_deficient`: Any natural number power of a prime is deficient.
+
+## Implementation Notes
+* Zero is not included in any of the definitions and these definitions only apply to natural
+  numbers greater than zero.
+
+## References
+* Prielipp, Robert W. “PERFECT NUMBERS, ABUNDANT NUMBERS, AND DEFICIENT NUMBERS.”
+  The Mathematics Teacher, vol. 63, no. 8, 1970, pp. 692–96. JSTOR,
+  http://www.jstor.org/stable/27958492. Accessed 21 Feb. 2024.
+
+## Tags
+
+abundant, deficient, weird, pseudoperfect
+-/
+
+open BigOperators Finset
+
+namespace Nat
+
+variable {n m p : ℕ}
+
+/-- `n : ℕ` is _abundant_ if the sum of the proper divisors of `n` is greater than `n`. -/
+def Abundant (n : ℕ) : Prop := ∑ i in properDivisors n, i > n
+
+/-- `n : ℕ` is _deficient_ if the sum of the proper divisors of `n` is less than `n`. -/
+def Deficient (n : ℕ) : Prop := ∑ i in properDivisors n, i < n
+
+/-- A positive natural number `n` is _pseudoperfect_ if there exists a subset of the proper
+  divisors of `n` such that the sum of that subset is equal to `n`. -/
+def Pseudoperfect (n : ℕ) : Prop :=
+  0 < n ∧ ∃ s ⊆ properDivisors n, ∑ i ∈ s, i = n
+
+/-- `n : ℕ` is a _weird_ number if and only if it is both abundant and pseudoperfect. -/
+def Weird (n : ℕ) : Prop := Abundant n ∧ ¬ Pseudoperfect n
+
+theorem not_pseudoperfect_iff_forall :
+    ¬ Pseudoperfect n ↔ n = 0 ∨ ∀ s ⊆ properDivisors n, ∑ i ∈ s, i ≠ n := by
+  rw [Pseudoperfect, not_and_or]
+  simp only [not_lt, nonpos_iff_eq_zero, mem_powerset, not_exists, not_and, ne_eq]
+
+theorem isDeficient_one : Deficient 1 := zero_lt_one
+
+theorem isDeficient_two : Deficient 2 := one_lt_two
+
+theorem isDeficient_three : Deficient 3 := by norm_num [Deficient]
+
+theorem isAbundant_twelve : Abundant 12 := by
+  rw [Abundant, show properDivisors 12 = {1,2,3,4,6} by rfl]
+  norm_num
+
+set_option maxRecDepth 1800 in
+theorem isWeird_seventy : Weird 70 := by
+  rw [Weird, Abundant, not_pseudoperfect_iff_forall]
+  have div70 : properDivisors 70 = {1, 2, 5, 7, 10, 14, 35} := by rfl
+  constructor
+  · rw [div70]
+    repeat norm_num
+  · rw [div70]
+    right
+    intro s hs
+    have hs' := mem_powerset.mpr hs
+    fin_cases hs' <;> decide
```

Comment:
I don't think these theorem names need to include `is`.

Reply:
I made them to include the `is` due to kbuzzard's comments above, but I can remove them.